pyfar/classes/coordinates.py

```diff
@@ -919,24 +1237,133 @@ def find_nearest_sph(self, points_1, points_2, points_3, distance,
             >>> coords = pf.samplings.sph_lebedev(sh_order=10)
             >>> result = coords.find_nearest_sph(0, 0, 1, 45, show=True)
         """
+        warnings.warn((
+            "This function will be deprecated in pyfar 0.7.0 in favor "
+            "of find_nearest_by_distance."),
+                PendingDeprecationWarning)
 
         # check the input
         assert distance >= 0 and distance <= 180, \
             "distance must be >= 0 and <= 180."
 
-        # get radius and check for equality
-        radius = self.get_sph()[..., 2]
-        delta_radius = np.max(radius) - np.min(radius)
-        if delta_radius > 1e-15:
-            raise ValueError(
-                "find_nearest_sph only works if all points have the same \
-                radius. Differences are larger than 1e-15")
+        # get target point in cartesian coordinates
+        coords = Coordinates(
+            points_1, points_2, points_3, domain, convention, unit)
 
         # get the points
         distance, index, mask = self._find_nearest(
-            points_1, points_2, points_3,
-            domain, convention, unit, show, distance, 'sph', atol,
-            np.max(radius))
+            coords, show, distance, 'sph', atol)
+
+        return index, mask
+
+    def find_nearest_points(self, coords, number_of_points=1):
+        """
+        Find the closest Coordinate point to given Points.
+        The search for the nearest point is performed using the scipy
+        cKDTree implementation..
+
+        Parameters
+        ----------
+        coords : Coordinates
+            Point to find nearest neighboring Coordinate
+        number_of_points : int, optional
+            Number of points to return. It must be > 0. The default is 1.
+
+        Returns
+        -------
+        distance : ndarray, double
+            Distance between the point and it's closest neighbor
+        index : int
+            Indexes of the closest points.
+        mask : boolean numpy array
+            mask that contains ``True`` at the positions of the selected points
+            and ``False`` otherwise. Mask is of shape ``cshape``.
+        """
+        # check the input
+        assert isinstance(number_of_points, int) and number_of_points > 0 and \
+            number_of_points <= self.csize,\
+            "number of points must be an integer > 0 and <= self.csize."
+
+        distance, index, mask = self._find_nearest(
+            coords, False, value=number_of_points, measure='k')
+
+        index = np.where(mask)
+        return distance, index, mask
+
+    def find_nearest_by_distance(self, coords, distance, measure='direct'):
```

Comment:
replaces `find_nearest_sph()` and `find_nearest_cart()`, measure defines which one is used. I was thinking this is a cleaner way of implementaion.

Reply:
And also removed the plot option, it is described in the example, how to do it if needed.